server/provider/scheduledMiniblocks.go

```diff
@@ -96,7 +107,10 @@ func findScheduledTransactionsHashes(block *data.Block) map[string]struct{} {
 	invalidTxs := make(map[string]struct{})
 
 	for _, miniblock := range block.MiniBlocks {
-		if miniblock.ProcessingType == string(Scheduled) {
+		hasProcessingTypeScheduled := miniblock.ProcessingType == dataBlock.Scheduled.String()
+		hasConstructionStateNotFinal := miniblock.ConstructionState != dataBlock.Final.String()
+
+		if hasProcessingTypeScheduled && hasConstructionStateNotFinal {
```

Comment:
might extract a new variable called `shouldAddToInvalidTxs`

Reply:
Fixed & renamed (the map previously had a wrong name).